### DISCUSSION TITLE
Changes to the duckdb_time_tz_struct

### DISCUSSION THREAD

**Discussion by Giorgi** [duckdb/duckdb#10917] [Ideas]
The pull request [fix(capi): add basic TIME_TZ support](https://github.com/duckdb/duckdb/pull/10207) added TimeTZ support to the C api but the `duckdb_time_tz_struct` that it introduces is inconsistent with other structs that exist in the C Api.

Let me explain it in more detail. Until that PR, for every date/time struct, there was a pair of structs - one that contained the raw underlying value and another that contained "decoded"  (human-readable) values. 

`duckdb_date` - Contains number of days since 1970-01-01
`duckdb_date_struct` - Mirror of the `duckdb_date` struct with days broken into year, month, day.

`duckdb_time` - Contains micros since 00:00:00
`duckdb_time_struct` - Mirror of the `duckdb_time` struct with hour, min, sec, and micros

`duckdb_timestamp` - Contains micros since 1970-01-01
`duckdb_timestamp_struct` - Mirror of `duckdb_timestamp` with date (that is `duckdb_date_struct` **and not** `duckdb_date`) and time (that is `duckdb_time_struct` **and not** `duckdb_time`)

To summarize, every struct that ends with `_struct` is a human-readable version of the same struct without `_struct`

But for TimeTZ this rule was broken:

```c
//! TIME_TZ is stored as 40 bits for int64_t micros, and 24 bits for int32_t offset
typedef struct {
	uint64_t bits;
} duckdb_time_tz;

typedef struct {
	duckdb_time time;
	int32_t offset;
} duckdb_time_tz_struct;
```

Given that this was added in 0.10.0 and 0.10.1 is scheduled in about two weeks, can we change `time` to `duckdb_date_struct`? I know that it's a breaking change but I doubt there is much code that uses this API and this change will improve the consistency of the API.

**Comment by SartorialOffense** [comment]
If it wasn't just introduced I would leave it. But given it just came out, I would make the change. It is just a small thing, but it is really nice to keep things consistent.

**Comment by Giorgi** [comment]
Implemented in https://github.com/duckdb/duckdb/pull/10933
